sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -1679,9 +1687,18 @@ alterColumnAction
     | dropDefault=DROP DEFAULT
     ;
 
+stringLitWithoutMarker
+    : STRING_LITERAL                                                                           #stringLiteralValue
+    | {!double_quoted_identifiers}? DOUBLEQUOTED_STRING                                        #doubleQuotedStringLiteralValue
+;
+
 stringLit
-    : STRING_LITERAL
-    | {!double_quoted_identifiers}? DOUBLEQUOTED_STRING
+    : stringLitWithoutMarker
+    | namedParameterMarkerVal
+    ;
+
+namedParameterMarkerVal
```

Comment:
This seems useless.

Reply:
Actually it is not. Because without pushing it down we need visits for   stringLitWithoutMarker (either all or none)